pilot/pkg/bootstrap/server.go

```diff
@@ -512,6 +513,14 @@ func (s *Server) WaitUntilCompletion() {
 	s.server.Wait()
 }
 
+func (s *Server) fetchPrivateKeyProviderConfig() *v1alpha1.PrivateKeyProvider {
+	meshConf := s.environment.Mesh()
+	proxyConf := meshConf.GetDefaultConfig()
```

Comment:
I think you may want to use `EffectiveProxyConfig` as there is a merge issue with meshConfig vs a proxy annotation.

Reply:
As far as I can tell this is not for Istiod, this is for responding to Gateway requests. Which still have per-node Proxyconfig we need to respect. So I agree with Jacob

Reply:
I mean istiod to gateways. I will check how can I use EffectiveProxyconfig.

Reply:
Found a reference usage [here](https://github.com/istio/istio/blob/master/pkg/kube/inject/webhook.go#L876). Updated the PR.